app/models/entry.rb

```diff
@@ -1,3 +1,42 @@
 class Entry < ApplicationRecord
   belongs_to :basket
+
+  EXEMPT_FROM_TAX = %w[chocolate book pills].freeze
+  BASIC_TAX_RATE = 0.1
+  IMPORT_TAX_RATE = 0.05
+  PRESITION = 20.0
+
+  belongs_to :basket
+
+  before_save :calulate_tax_update_price
+
+  private
+
+  def calulate_tax_update_price
+    self.basic_tax = calcul_basic_tax
+    self.import_tax = calcul_import_tax
+    self.final_price = calcul_final_price
+  end
+
+  def calcul_basic_tax
+    return round_up(price.to_f.round(2) * BASIC_TAX_RATE) unless EXEMPT_FROM_TAX.any? do |exception|
```

Comment:
Maybe in one line without the block this could be `unless EXEMPT_FROM_TAX.include? name`?